test/e2e/suites/chart-upgrade/chart_upgrade_test.go

```diff
@@ -152,164 +148,106 @@ var _ = Describe("Chart upgrade functionality should work", Ordered, Label(e2e.S
 				},
 			}
 		})
-	})
 
-	It("Should upgrade Turtles to head/main and validate providers", func() {
-		// There can be only one core CAPI provider installed at a time.
-		By("Remove the core CAPI provider before upgrade")
-		testenv.RemoveCAPIProvider(ctx, testenv.RemoveCAPIProviderInput{
+		By("Uninstalling Turtles v0.24.3 (providers and workload cluster keep running)")
+		testenv.UninstallRancherTurtles(ctx, testenv.UninstallRancherTurtlesInput{
 			BootstrapClusterProxy: bootstrapClusterProxy,
-			ProviderName:          "cluster-api",
-			ProviderNamespace:     "capi-system",
+			DeleteWaitInterval:    e2eConfig.GetIntervals(bootstrapClusterProxy.GetName(), "wait-controllers"),
 		})
 
-		// Upgrade Turtles chart to locally built one
-		testenv.DeployRancherTurtles(ctx, testenv.DeployRancherTurtlesInput{
+		By("Patching Turtles CRDs with Helm release annotations for cattle-turtles-system (Step 3 of migration guide)")
+		framework.RunCommand(ctx, framework.RunCommandInput{
```

Comment:
can you leave a link to the Turtles docs somewhere, since we reference migration guide here?